source/transfer_list.rst

```diff
@@ -902,5 +902,172 @@ This manifest contains the SPMC attribute node consumed by the SPMD
      - hdr_size
      - Holds a SPMC manifest image in DT format.
 
+
+**Executable entry point information entry layout (XFERLIST_EXEC_EP_INFO)**
+
+This entry type holds one or more `entry_point_info_t` structures.
+
+`entry_point_info_t` is a TF-A-specific data structure [TF_BL31]_ used to
+represent the execution state of an image; that is, the state of general purpose
+registers, PC, and SPSR. Due to architectural differences in AArch64 and
+AArch32, there exist two representations of `entry_point_info_t`.
+
+.. entry_point_info_t:
+.. list-table:: entry_point_info_t structure layout (AArch64)
+   :widths: 2 2 2 8
+
+   * - field
+     - size (bytes)
+     - offset (bytes)
+     - description
+
+   * - h
+     - 0x8
+     - 0x0
+     - Structure header `param_header_t`.
+
+   * - pc
+     - 0x8
+     - 0x8
+     - Program counter (PC) pointing to the entry point of the executable.
+
+   * - spsr
+     - 0x4
+     - 0xc
+     - Holds the saved process state (SPSR) for the next executable.
+
+   * - args
+     - 0x40
+     - 0x18
+     - Holds the values of registers `x0` to `x7`.
+
+.. entry_point_info_t:
+.. list-table:: entry_point_info_t structure layout (AArch32)
+   :widths: 2 2 2 8
+
+   * - Field
+     - Size (bytes)
+     - Offset (bytes)
+     - Description
+
+   * - h
+     - 0x8
+     - 0x0
+     - Header `param_header_t` holding information about the type, size, version and attributes of
+       this structure.
+
+   * - pc
+     - 0x4
+     - 0x8
+     - Program counter pointing to the entry point of the executable.
+
+   * - spsr
+     - 0x4
+     - 0xc
+     - Holds the saved process state for the next executable.
+
+   * - lr_svc
+     - 0x4
+     - 0x10
+     - Holds the link register (LR) value used during exception return in Secure
+       or Non-secure Supervisor mode.
+
+   * - args
+     - 0x40
+     - 0x14
+     - Holds the values of registers `r0` to `r3`.
+
+The header of `entry_point_info_t` holds information about the attributes of the
+executable. Bits 0 and 5 of this field are used to encode the security state;
+in other words, whether the image is to be executed in Secure, Non-Secure, or Realm
+mode.
+
+.. param_header_t:
+.. list-table:: Layout of header data structure `param_header_t`.
+
+   * - Field
+     - Size (bytes)
+     - Offset (bytes)
+     - Description
+
+   * - type
+     - 0x1
+     - 0x0
+     - Type of the represented structure i.e. executable info.
+
+   * - version
```

Comment:
We should not need a version within a TE. Can we drop this field?

Reply:
Is this intended for use by multiple projects, or just one? Given that things are moving to firmware handoff, would that not be an opportunity to review the structure, since the generator and any clients would have to change anyway?

But if it is a pre-existing binary structure that we have no control over, then yes I suppose we just have to live with it.

Even so, we do need to think about the 32/64-bit side of things, as I commented elsewhere.